ts/sagas/authentication.ts

```diff
@@ -105,6 +110,43 @@ export function* watchSessionExpired(): IterableIterator<Effect> {
   }
 }
 
+export function* watchLogoutRequest(): IterableIterator<Effect> {
+  while (true) {
+    yield take(LOGOUT_REQUEST);
+
+    // Get the SessionToken from the store
+    const sessionToken: SessionToken | undefined = yield select(
+      sessionTokenSelector
+    );
+
+    if (sessionToken) {
+      const backendClient = BackendClient(apiUrlPrefix, sessionToken);
+
+      // Call the Backend service
+      const response:
+        | BasicResponseTypeWith401<SuccessResponse>
+        | undefined = yield call(backendClient.logout, {});
+
+      if (!response || response.status !== 200) {
+        // We got a error, send a LOGOUT_FAILURE action
```

Comment:
this is tricky, because if we get a 401, our session is already invalid, so we're basically already logged out

instead, if we issue a "logout error" bug we keep the user logged in, the user will never be able to logout!

Reply:
Ok for the 401, whe can dispatch logoutFailure only if the response status is a error but is different from 401 (in that case logout is not needed because as you said the user is already logged out with SESSION_EXPIRED).

For the "logout error" how you think we can deal? What if the user wants to logout but the Backed doesn't answer 200? I think is better send a message in Mixpanel but logout the user anyway. The "not destroyed" session of the Backend will for sure be deleted when the user with the same CF try to login again (for the only one session active rule), or auto-expire after the "expire time".

Reply:
yes I agree, as soon as the app destroys the session token, the user is effectively logged out - the session in the backend will eventually expire

so I think we should logout the user from the app even if the backend responds with an error

Reply:
eventually when we implement the "auto logout from existing sessions on login" the backend will be forced to garbage collect the "zombie" sessions...